### PR TITLE
Update ParentRef wording

### LINKED ISSUE #973: Route ParentRef field does not mention ReferencePolicy
**What happened**:

The `Namespace` field of the `ParentRef` structure doesn't mention whether a `ReferencePolicy` is required for the controller to follow a cross-namespace reference.

**What you expected to happen**:

I would expect all cross-namespace references to check a `ReferencePolicy`.

### PR DESCRIPTION
**Pull request by youngnick**
**What type of PR is this?**
/kind cleanup
/kind documentation

**What this PR does / why we need it**:
Notes that ParentRefs that cross namespace boundaries must have some way for the referent to accept/allow the routes.

**Which issue(s) this PR fixes**:

Fixes #973

**Does this PR introduce a user-facing change?**:
```release-note
Clarified that AllowedRoutes or ReferenceGrant must be used by a parent in a cross-namespace ParentRef.
```

### CODE REVIEW

**Comment by youngnick** [comment]
I didn't update the v1alpha2 docs; happy to if we think that's worthwhile for this change.

File: apis/v1beta1/shared_types.go [hunk 1]
```diff
@@ -140,6 +140,13 @@ type CommonRouteSpec struct {
 	// case, the list of routes attached to those resources should also be
 	// merged.
 	//
+	// Note that for ParentRefs that cross namespace boundaries, the parent
+	// being referred to must have some mechanism to allow Routes to attach.
+	// Gateway has the AllowedRoutes field, for example. Any other resource
+	// being used as a parent across namespaces MUST either include an AllowedRoutes
+	// field or equivalent functionality, OR use a ReferenceGrant (if the spec
+	// can't have a field added).
```
Comment (robscott) [inline apis/v1beta1/shared_types.go:148]: This is a great clarification, a few nits though:

1. This feels like something that belongs on the Namespace field within the ParentRef type definition.
2. As you mentioned, I think it would be good to include this in both v1a2 and v1b1 type definitions. They should be kept identical.
3. Maybe reword this to say something like "Cross-namespace references are only valid if they are explicitly allowed by something in the namespace they are referring to. For example, Gateway has the AllowedRoutes field, and ReferenceGrant provides a generic way to enable any other kind of cross-namespace reference."

**Review by robscott — COMMENTED** [review]
Thanks @youngnick!

File: apis/v1beta1/shared_types.go (hunk 1 above)
Comment (youngnick) [reply apis/v1beta1/shared_types.go:148]: Great wording, thanks. I'll update.

**Comment by youngnick** [comment]
Rebased and updated, PTAL.

**Comment by robscott** [comment]
Thanks @youngnick! LGTM once tests pass.

File: apis/v1alpha2/shared_types.go [hunk 2]
```diff
@@ -25,6 +25,12 @@ import (
 // with "Core" support is Gateway. This API may be extended in the future to
 // support additional kinds of parent resources, such as HTTPRoute.
 //
+// Note that for ParentRefs that cross namespace boundaries, there are specific
+// rules. Cross-namespace references are only valid if they are explicitly
+// allowed by something in the namespace they are referring to. For example,
+// Gateway has the AllowedRoutes field, and ReferenceGrant provides a
+// generic way to enable any other kind of cross-namespace reference.
```
Comment (shaneutt) [inline apis/v1alpha2/shared_types.go:32]:
```suggestion
// Note that for ParentRefs that cross namespace boundaries there are specific
// rules. Cross-namespace references are only valid if they are explicitly
// allowed by something in the namespace they are referring to. For example:
// Gateway has the AllowedRoutes field, and ReferenceGrant provides a
// generic way to enable any other kind of cross-namespace reference.
```
